src/sentry/tasks/deletion.py

```diff
@@ -267,6 +267,52 @@ def delete_group(object_id, transaction_id=None, **kwargs):
         )
 
 
+@instrumented_task(
+    name='sentry.tasks.deletion.delete_groups',
+    queue='cleanup',
+    default_retry_delay=60 * 5,
+    max_retries=MAX_RETRIES
+)
+@retry(exclude=(DeleteAborted, ))
+def delete_groups(object_ids, transaction_id=None, eventstream_state=None, **kwargs):
+    from sentry import deletions, eventstream
+    from sentry.models import Group
+
+    if not object_ids:
+        return
+
+    object_ids = sorted(object_ids)
+    group_id = object_ids[0]
+
+    task = deletions.get(
+        model=Group,
+        query={
+            'id': group_id,
+        },
+        transaction_id=transaction_id or uuid4().hex,
+    )
+    has_more = task.chunk()
+    if has_more:
+        countdown = 15
+    else:
+        # *this* group is done, remove it from the list
+        object_ids.pop(0)
+        # we're moving on to the next group, don't delay
+        countdown = None
```

Comment:
Also, `countdown=None` should put a task into the queue for immediate processing. I don't know the overhead of that, but we could do `N` different groups (instead of 1, as currently written) per `delete_groups` call before issuing a new task if we feel that the task-hopping will slow down batch deletes too much?

Reply:
> The existing `delete_group` would add a countdown delay of 15 if there was more to chunk, but we don't want to put a 15 second delay between every group in the batch.

We may have already been over this, but what was the 15 second delay for originally? Shouldn't these naturally interleave themselves as the queue is processed? I don't think it'd actually throttle deletion throughput (it doesn't ensure we're doing less work), it'd just add 15 seconds to every deletion…?

> Also, `countdown=None` should put a task into the queue for immediate processing. I don't know the overhead of that, but we could do `N` different groups (instead of 1, as currently written) per `delete_groups` call before issuing a new task if we feel that the task-hopping will slow down batch deletes too much?

Paging @JTCunning…

Reply:
I can't tell you what the 15s countdown is for other than just "is probably for safe". It's a standard throughout all of deletion tasks right now and was established in 2014 (https://github.com/getsentry/sentry/commit/ad7d853ca16f8cd89d70510bbd021069ee609bf1).

Reply:
Hmm, I wonder if we can remove the delay for chunks *within* an existing delete...

@JTCunning any thoughts on the second part Ted quoted?

Reply:
Now that I take a look at all of this, why are we deleting all of these serially instead of using the `delete_bulk` method in sentry.deletions? It still returns the `has_more` boolean and chunks correctly.

We should leverage it so that we can hit the `mark_deletion_in_progress` method as fast as possible.

Reply:
OK, updated to use `id__in` (which will do batch deletes) and also do max 100 per batch (as discussed in Slack to avoid huge `IN` queries).